src/laboratory/api/views.py

```diff
@@ -702,3 +703,199 @@ def list(self, request, *args, **kwargs):
         self.org_pk = kwargs['org_pk']
         self.lab_pk = kwargs['lab_pk']
         return super().list(request, *args, **kwargs)
+
+
+class ReactiveManagementViewset(AuthAllPermBaseObjectManagement):
+    serializer_class = {
+        'list': serializers.ReactiveDataTableSerializer,
+        'destroy': serializers.ReactiveSerializer,
+        'create': serializers.ValidateReactiveSerializer,
+        'update': serializers.ValidateReactiveSerializer
+    }
+    perms = {
+        'list': ["laboratory.view_object"],
+        'create': ["laboratory.add_object", "laboratory.view_object"],
+        'update': ["laboratory.change_object", "laboratory.view_object"],
+        'destroy': ["laboratory.delete_object", "laboratory.view_object"]
+    }
+
+    permission_classes = (PermissionByLaboratoryInOrganization,)
+
+    queryset = Object.objects.filter(type=Object.REACTIVE)
+    pagination_class = LimitOffsetPagination
+    filter_backends = (DjangoFilterBackend, SearchFilter, OrderingFilter)
+    search_fields = ['code', 'name']  # for the global search
+    filterset_class = filterset.ReactiveFilter
+    ordering_fields = ['code']
+    ordering = ('code',)  # default order
+    operation_type = ''
+    org_pk, lab_pk, org = None, None, None
+
+    def get_response_validate_data(self, reactive_serializer, reactive_ch_serializer):
+        reactive_changed_data = list(
+            reactive_serializer.validated_data.keys())
+        reactive_ch_changed_data = list(
+            reactive_ch_serializer.validated_data.keys())
+
+        # Multiple response data
+        response_data = reactive_serializer.data
+        reactive_ch_data = reactive_ch_serializer.data
+
+        # THIS ID SHOULDN'T REPLACE THE MAIN ID(EQUIPMENT OBJECT)
+        del reactive_ch_data['id']
+        response_data.update(reactive_ch_data)
+
+        return response_data, reactive_changed_data, reactive_ch_changed_data
+
+    def get_reactive_ch_serializer(self, instance, request, partial, lab_pk):
+        if hasattr(instance, 'sustancecharacteristics'):
+            reactive_ch_instance = instance.sustancecharacteristics
+            reactive_ch_serializer = ValidateReactiveCharacteristicsSerializer(
+                reactive_ch_instance, data=request.data, partial=partial, context={"lab_pk": lab_pk})
```

Comment:
Why you are sending a lab_pk parameter by context if you are not using in ValidateReactiveCharacteristicsSerializer.